File: ai/src/main/java/org/conductoross/conductor/ai/models/ChatMessage.java
package org.conductoross.conductor.ai.models;

import java.util.ArrayList;
import java.util.List;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class ChatMessage {

    public enum Role {
        user,
        assistant,
        system,
        // When chat completes requests execution of tools
        tool_call,

        // Actual tool execution and its output
        tool
    }

    private Role role;
    private String message;
    private List<String> media = new ArrayList<>();
    private String mimeType;
    private List<ToolCall> toolCalls;

    public ChatMessage(Role role, String message) {
        this.role = role;
        this.message = message;
    }

    public ChatMessage(Role role, ToolCall toolCall) {
        this.role = role;
        this.toolCalls = List.of(toolCall);
    }
}

File: ai/ai/src/models/ChatMessage.ts
import { ToolCall } from './ToolCall.js';

export enum ChatMessageRole {
  user = 'user',
  assistant = 'assistant',
  system = 'system',
  /** When chat completes requests execution of tools */
  tool_call = 'tool_call',
  /** Actual tool execution and its output */
  tool = 'tool',
}

export class ChatMessage {
  role: ChatMessageRole;
  message?: string;
  media: string[] = [];
  mimeType?: string;
  toolCalls?: ToolCall[];

  constructor(role: ChatMessageRole, messageOrToolCall?: string | ToolCall) {
    this.role = role;
    if (typeof messageOrToolCall === 'string') {
      this.message = messageOrToolCall;
    } else if (messageOrToolCall) {
      this.toolCalls = [messageOrToolCall];
    }
  }
}
